cve_bin_tool/extractor.py

```diff
@@ -109,6 +113,43 @@ async def extract_file_rpm(self, filename, extraction_path):
                     return 1
         return 0
 
+    async def extract_file_zst(self, filename, extraction_path):
+        """Extract zstd compressed files"""
+
+        dctx = zstandard.ZstdDecompressor()
+        with ErrorHandler(mode=ErrorMode.Ignore) as e:
+            with open(filename, "rb") as f:
+                with dctx.stream_reader(f) as reader:
+                    with tarfile.open(mode="r|", fileobj=reader) as tar:
+                        tar.extractall(extraction_path)
+                    tar.close()
+        return e.exit_code
+
+    async def extract_file_pkg(self, filename, extraction_path):
+        """Extract pkg files"""
+        if sys.platform.startswith("linux"):
+            if not await aio_inpath("tar"):
+                with ErrorHandler(mode=self.error_mode, logger=self.logger):
+                    raise Exception("tar is required to extract .pkg files")
+            else:
+                stdout, stderr, return_code = await aio_run_command(
+                    ["tar", "xf", filename, "-C", extraction_path]
```

Comment:
Hey @terriko 
I gave it another shot, and `tarfile.open()` doesn't seem to give a usable solution.
Tarfile can't open .pkg files directly, and since pkg files are essentially `txz archives`, just changing the package extension to append `.xz` does the trick for a few packages. But while it works on a few, the same function fails to extract other .pkg files. Looks the older and newer versions of FreeBSD seem to use different compressions. This inconsistent behavior is the reason I used the `tar` command.
Maybe we can merge this for now and look for a possible solution in the future?

Reply:
Can you put a comment explaining your results into the code directly so anyone wanting to pick it up later can understand what you've tried already? It's too easy to lose track of great research like this if it only lives in the pull request!